internal/pkg/cli/app_package_test.go

```diff
@@ -2,3 +2,293 @@
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Times(0)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+			},
+
+			wantedErrorS: "there are no applications in the workspace, run `archer init` first",
+		},
+		"wrap list envs error": {
+			inAppName: "frontend",
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Return(nil, errors.New("some ssm error"))
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+			},
+
+			wantedAppName: "frontend",
+			wantedErrorS:  "list environments for project : some ssm error",
+		},
+		"prompt for all options": {
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Return([]string{"frontend", "backend"}, nil)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Return([]*archer.Environment{
+					{
+						Name: "test",
+					},
+					{
+						Name: "prod",
+					},
+				}, nil)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which application's CloudFormation template would you like to generate?", gomock.Any(), []string{"frontend", "backend"}).Return("frontend", nil)
+				m.EXPECT().SelectOne("Which environment's configuration would you like to use for your stack's parameters?", gomock.Any(), []string{"test", "prod"}).Return("test", nil)
+			},
+
+			wantedAppName: "frontend",
+			wantedEnvName: "test",
+		},
+		"prompt only for the app name": {
+			inEnvName: "test",
+
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Return([]string{"frontend", "backend"}, nil)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Times(0)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which application's CloudFormation template would you like to generate?", gomock.Any(), []string{"frontend", "backend"}).Return("frontend", nil)
+			},
+
+			wantedAppName: "frontend",
+			wantedEnvName: "test",
+		},
+		"prompt only for the env name": {
+			inAppName: "frontend",
+
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Return([]*archer.Environment{
+					{
+						Name: "test",
+					},
+					{
+						Name: "prod",
+					},
+				}, nil)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which environment's configuration would you like to use for your stack's parameters?", gomock.Any(), []string{"test", "prod"}).Return("test", nil)
```

Comment:
Again might be good to constantize prompts to make tests more maintainable

Reply:
:+1: good idea! Moved the prompts to constants